Feat/ login request

byungchanKo99 opened the pull request:
로그인 기능 만들었습니다.
- login context : provider를 최상단 app.tsx에서 감싸줬습니다. 다른 예시를 찾아보니 consumer 대신 context 값을 useState로 만들어서 쓰던데 괜찮나요...? Context값은 관리자용에선 직위만 가지고 있으면 될 것 같은데 더 필요한 것 있을까요...?
- 로그인페이지에서 Post 요청 후 성공하면 직위를 context에 넣고, headers에 'x-access-token' 을 키 값으로 token값을 넣어줬습니다. 그 후 `HomeNavigator`로 이동하게 했습니다.
- 실패하면 실패 모달창을 띄우게 했습니다.

byungchanKo99 committed:
feat: constant api

byungchanKo99 committed:
feat:/login context

children type을 도통 뭐로해야할지 모르겠어서 일단 any로 해뒀습니다.

byungchanKo99 committed:
feat/ loginProvider

byungchanKo99 committed:
feat/ 비밀번호 입력 안보이게

byungchanKo99 committed:
feat/ signin POST

byungchanKo99 committed:
feat/ loginContext 사용

byungchanKo99 committed:
feat/로그인 실패 모달 연결

byungchanKo99 committed:
feat/ lint error

DonghyunKim98 commented on app/components/screens/signIn/index.tsx:
```diff
@@ -95,7 +100,29 @@ function SignIn() {
+        id,
+        password: pw,
+      })
+      .then((res: any) => {
```
- any 를 쓸바엔 차라리 typing 을 하지 맙시다.

DonghyunKim98 commented on app/components/screens/signIn/index.tsx:
```diff
@@ -95,7 +100,29 @@ function SignIn() {
 
   const signInBtnClickListener = () => {
-    navigation.navigate('HomeNavigator');
+    console.log(`${id}, ${pw}`);
```
- 콘솔 지워주세요

DonghyunKim98 commented on app/components/screens/signIn/index.tsx:
```diff
@@ -95,7 +100,29 @@ function SignIn() {
+        const {accessToken, position} = res.data;
+
+        login(position);
+        console.log(JSON.stringify(res, null, 2));
```
- 콘솔지워주세요 ㅠㅠ

DonghyunKim98 commented on app/components/screens/signIn/index.tsx:
```diff
@@ -95,7 +100,29 @@ function SignIn() {
+      .finally(() => {
+        setId('');
+        setPw('');
+      });
```
- 필요한건가요? 에러가 발생할때만 input 을 비우는 것 같은데.. catch 문안에 넣어주세요.
- 그리고 계속해서 말하지만, 이런식으로 적어두면 무슨 역할을 하는지 알수 없습니다.
    - `cleanUpInput` 등의 함수를 만드는건 어떤가요?

DonghyunKim98 commented on app/components/screens/signIn/index.tsx:
```diff
@@ -95,7 +100,29 @@ function SignIn() {
+        navigation.navigate('HomeNavigator');
+      })
+      .catch((err: any) => {
+        console.log(JSON.stringify(err));
```
- 콘솔 찍을때 `err.response` 를 찍으면 더 수월하게 에러 처리를 할 수 있습니다.
- 참고 사이트 : https://xn--xy1bk56a.run/axios/guide/error-handling.html

DonghyunKim98 commented on app/utils/context/LoginContext.tsx:
```diff
@@ -0,0 +1,20 @@
+      {children}
+    </LoginContext.Provider>
+  );
+};
```
- 전반적인 기능을 보았을땐 차라리 `userInfoProvider` 에 더 적합한 것 같습니다.
- property 보단 user 로 명시를 하고 객체를 받아서 할당하는게 더 나을것 같아요.
```tsx
interface userInterface {
  position: string;
  name: string;
}
//
//
//
export const UserProvider = ({children}: ReactNode) => {
  const [user, setUser] = useState<userInterface>('');

  return (
    <LoginContext.Provider value={{user, setUser}}>
      {children}
    </LoginContext.Provider>
  );
};
``` 
- 이렇게 간단하게 user 와 setUser 만 제공을 하고 setUser 를 이용하여 Login 을 진행하는 페이지에서 실제로 유저를 바꾸는게 좋아보입니다.

byungchanKo99 committed:
feat: 입력값 지우는 함수

byungchanKo99 committed:
feat: api 파일 폴더로 옮김

byungchanKo99 committed:
feat: console log 삭제

byungchanKo99 committed:
fix: api instance에 토큰 넣기

byungchanKo99 committed:
feat: context 수정

이름 변경 : LoginContext -> UserInfoContext
로그인했을 때 response가 직위밖에 안와서 일단 user state는 그대로 string으로 했습니다.
반환값 : 기존 login 함수 삭제, setUser로 변경.

byungchanKo99 committed:
fix: lint error
